src/Orleans/Async/TaskExtensions.cs

```diff
@@ -141,7 +141,7 @@ public static Task<object> Box(this Task<object> task)
         private static Task<object> TaskFromFaulted(Task task)
         {
             var completion = new TaskCompletionSource<object>();
-            completion.SetException(task.Exception);
+            completion.SetException(task.Exception.InnerException);
```

Comment:
`Exception.InnerException` handles is ok there, but as sanity check this can be rewritten in the following way:  
`try { task.GetAwaiter().GetResult() 
}`  ..